modin/core/execution/unidist/implementations/pandas_on_unidist/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,566 @@
+
+        Returns
+        -------
+        PandasOnUnidistDataframeVirtualPartition
+            An axis partition containing only a single materialized partition.
+        """
+        materialized = super(
+            PandasOnUnidistDataframeVirtualPartition, self
+        ).force_materialization(get_ip=get_ip)
+        self._list_of_block_partitions = materialized.list_of_block_partitions
+        return materialized
+
+    def mask(self, row_indices, col_indices):
+        """
+        Create (synchronously) a mask that extracts the indices provided.
+
+        Parameters
+        ----------
+        row_indices : list-like, slice or label
+            The row labels for the rows to extract.
+        col_indices : list-like, slice or label
+            The column labels for the columns to extract.
+
+        Returns
+        -------
+        PandasOnUnidistDataframeVirtualPartition
+            A new ``PandasOnUnidistDataframeVirtualPartition`` object,
+            materialized.
+        """
+        return (
+            self.force_materialization()
+            .list_of_block_partitions[0]
+            .mask(row_indices, col_indices)
+        )
+
+    def to_pandas(self):
+        """
+        Convert the data in this partition to a ``pandas.DataFrame``.
+
+        Returns
+        -------
+        pandas DataFrame.
+        """
+        return self.force_materialization().list_of_block_partitions[0].to_pandas()
+
+    _length_cache = None
+
+    def length(self):
+        """
+        Get the length of this partition.
+
+        Returns
+        -------
+        int
+            The length of the partition.
+        """
+        if self._length_cache is None:
+            if self.axis == 0:
+                self._length_cache = sum(
+                    obj.length() for obj in self.list_of_block_partitions
+                )
+            else:
+                self._length_cache = self.list_of_block_partitions[0].length()
+        return self._length_cache
+
+    _width_cache = None
+
+    def width(self):
+        """
+        Get the width of this partition.
+
+        Returns
+        -------
+        int
+            The width of the partition.
+        """
+        if self._width_cache is None:
+            if self.axis == 1:
+                self._width_cache = sum(
+                    obj.width() for obj in self.list_of_block_partitions
+                )
+            else:
+                self._width_cache = self.list_of_block_partitions[0].width()
+        return self._width_cache
+
+    def drain_call_queue(self, num_splits=None):
+        """
+        Execute all operations stored in this partition's call queue.
+
+        Parameters
+        ----------
+        num_splits : int, default: None
+            The number of times to split the result object.
+        """
+        if len(self.call_queue) == 0:
+            # this implicitly calls `drain_call_queue`
```

Comment:
this comment looks weird, I read it as "this implicitly falls into infinite recursion" (which should not be the case as tests won't pass) 🙃 please explain or fix the comment

Reply:
list_of_blocks -> _list_of_block_partitions (this is PandasOnUnidist partitions) -> list_of_blocks so there is no infinite recursion

Reply:
nope, your comment says "drain_call_queue() -> length of queue is zero -> implicitly calls drain_call_queue()", this sounds like an infinite recursion to me

Reply:
elaborated the comment